src/posix.h

```diff
@@ -89,6 +89,12 @@
 #define EAFNOSUPPORT (INT_MAX-1)
 #endif
 
+/* Compiler independent macro to handle signal interrpted system calls */
+#define HANDLE_EINTR(result, x) do {					\
```

Comment:
We tend to avoid macros to do things like this...  again, my preference is generally to avoid having to jump around too much within the code base to understand what a piece of code is doing.  I don't love macros in general, and in particular, I don't like macros that loop (or do flow control).

Again, there are definitely exceptions in our codebase (there's some `vector_foreach` type of macro that is in pretty regular use).  But I think that something that's used just twice is a place where we can safely repeat ourselves without too much downside.  It think that the clarity outweighs the lack of DRYness here.

This is mostly personal opinion, and again, I don't think that it's so important that we shouldn't merge this.  The benefits from this PR far outweigh any macro-related gripes that I have.  😁   Just trying to convey some stylistic thoughts.